### DISCUSSION TITLE
Unable to build DBeaver using Eclipse

### DISCUSSION THREAD

**Discussion by stegus64** [dbeaver/dbeaver#8627] [Q&A] [answered]
This is the first time I try building a Maven project using Eclipse. I am probably doing something simple wrong.

I have tried to follow the instructions in the DBeaver Wiki.

I have installed Maven and the exact Eclipse version mentioned in the Wiki.

The command-line Maven build works as expected.

When I try try to import the Maven project I get the following problems:

![image](https://user-images.githubusercontent.com/51655419/81069106-bdb89d80-8ee1-11ea-9334-e4b259cd6d52.png)

When Eclipse tries to build, I get build errors in almost every package, 113853 errors according to the problems window.

![image](https://user-images.githubusercontent.com/51655419/81069348-0cfece00-8ee2-11ea-88cf-101f67f773b5.png)

#### System information: 
- Operating system (distribution) and version
Windows 10 64-bit

Any suggestions?

**Comment by deceits** [comment]
Hi Stegus64, 
I also ran into the same issue. I believe you need to install additional Eclipse plugins which DBeaver depends on as described in wiki of Develop in Eclipse.

Hope this helps,

**Comment by stegus64** [comment]
I have followed the instructions in the Wiki exactly. I have already installed all plugins mentioned in the Wiki.

**Comment by tommai78101** [comment]
Question: Are you installing DBeaver on the latest Eclipse release, 2020-06?

DBeaver fails to install through the Eclipse Marketplace on 2020-06, but it installs perfectly fine on 2020-03, so I'm wondering if this issue is related.

**Comment by stegus64** [comment]
I am not trying to install DBEaver as an Eclipse plugin.
I am trying to build DBeaver from sources using Eclipse as described in the wiki:
https://github.com/dbeaver/dbeaver/wiki/Develop-in-Eclipse
As recommended in the Wiki I use Eclipse 2018-09 https://www.eclipse.org/downloads/packages/release/2018-09/r/eclipse-ide-rcp-and-rap-developers

**Comment by serge-rider** [comment]
@stegus64 
What Eclipse IDE do you use? (packaging and version).

Maybe it is one of the known Eclipse IDE bugs. It needs Tycho plugins in order to build DBeaver. Eclipse installs Tycho extension automatically when you import/create some Tycho based plugin.
Perhaps you use a new Eclipse instance with older workspace (made by some other Eclipse instance) so Eclipse don't have Tycho extension installed.

Workaround: delete some plugin (only Eclipse project, not files) and then import it again. Eclipse will ask about Tycho extension install - just do it and this will resolve the issue.

**Comment by stegus64** [comment]
This is a snapshot from my About dialog:
![image](https://user-images.githubusercontent.com/51655419/81779699-89eb0280-94f5-11ea-92e5-279169de9374.png)
As recommended in the Wiki I use Eclipse 2018-09
https://www.eclipse.org/downloads/packages/release/2018-09/r/eclipse-ide-rcp-and-rap-developers

My computer is Windows 10 x64

I think I might have had a very old version of Eclipse installed on this machine several years ago. Maybe that is causing problems? Can I somehow make sure I get a clean start?

> Workaround: delete some plugin (only Eclipse project, not files)

Exactly what do you mean by this? Please remember that this is the first time I try to use Eclipse for something complex.

**Comment by stegus64** [comment]
I tried creating a completely new workspace and importing the dbeaver project again. I get this dialog:
![image](https://user-images.githubusercontent.com/51655419/81780968-ad16b180-94f7-11ea-8285-071ef3339888.png)
As you see there is no option to install tycho. Only to ignore it.
I do not understand why Eclipse does not install tycho automatically as it is supposed to...

**Comment by tommai78101** [comment]
> I do not understand why Eclipse does not install tycho automatically as it is supposed to... 

I think it's telling you that there are 102 errors found, which I think it's a Maven POM or Gradle error? Hence it doesn't install Tycho automatically.

**Comment by serge-rider** [comment, marked as answer]
I would recommend to use most recent Eclipse (2018-09 is really old).
We support most of Eclipse 4 versions including really old ones but only on binary level. If you want to build from sources you need to have fresh version.
